File: Sample/app/src/main/java/com/example/workspace/sample/pages/top/HomeViewModelImpl.java
package com.example.workspace.sample.pages.top;

import android.util.Log;
import android.view.View;

import com.example.workspace.sample.R;

public class HomeViewModelImpl extends HomeViewModel {
    private HomeView homeView;

    @Override
    public void init(HomeView view) {
        this.homeView = view;
    }

    @Override
    public void onClick(View view) {
    }
}

File: Sample/app/src/main/java/com/example/workspace/sample/pages/top/HomeViewModelImpl.kt
package com.example.workspace.sample.pages.top

import android.view.View

class HomeViewModelImpl : HomeViewModel() {
    private var homeView: HomeView? = null

    override fun init(view: HomeView) {
        this.homeView = view
    }

    override fun onClick(view: View) {
        view.scaleX = 2f
        view.scaleY = 2f
    }
}
